### ISSUE TITLE
Issue with upgrading contracts with enum value as a key in a mapping

### ISSUE THREAD

**Issue by NarkomDe** [OpenZeppelin/openzeppelin-upgrades#300]
I have a contract with a mapping from an enum value to another struct in a storage. I've encountered that when trying to upgrade it(even to the contract with the same code), with stack-trace:
```
     Error: An unexpected condition occurred. Please report this at https://zpl.in/upgrades/report
      at Object.assert (node_modules/@openzeppelin/upgrades-core/src/utils/assert.ts:3:11)
      at StorageLayoutComparator.uncachedGetTypeChange (node_modules/@openzeppelin/upgrades-core/src/storage/compare.ts:192:9)
      at StorageLayoutComparator.getTypeChange (node_modules/@openzeppelin/upgrades-core/src/storage/compare.ts:111:27)
      at StorageLayoutComparator.getFieldChange (node_modules/@openzeppelin/upgrades-core/src/storage/compare.ts:83:29)
      at node_modules/@openzeppelin/upgrades-core/src/storage/compare.ts:72:63
      at change (node_modules/@openzeppelin/upgrades-core/src/levenshtein.ts:77:20)
      at buildMatrix (node_modules/@openzeppelin/upgrades-core/src/levenshtein.ts:46:29)
      at Object.levenshtein (node_modules/@openzeppelin/upgrades-core/src/levenshtein.ts:16:18)
      at StorageLayoutComparator.layoutLevenshtein (node_modules/@openzeppelin/upgrades-core/src/storage/compare.ts:72:17)
      at StorageLayoutComparator.compareLayouts (node_modules/@openzeppelin/upgrades-core/src/storage/compare.ts:64:47)
      at Object.assertStorageUpgradeSafe (node_modules/@openzeppelin/upgrades-core/src/storage/index.ts:19:29)
      at prepareUpgradeImpl (node_modules/@openzeppelin/truffle-upgrades/src/upgrade-proxy.ts:41:3)
      at upgradeProxy (node_modules/@openzeppelin/truffle-upgrades/src/upgrade-proxy.ts:80:20)
      at Context.<anonymous> (test/SomeTestName.js:654:17)
```
Commenting out node_modules/@openzeppelin/upgrades-core/src/storage/compare.ts:192 line helps, upgrade go as expected and upgraded contract is working fine.But I think there is a better option to solve this problem that plainly commenting out checks. :)

**Comment by frangio** [comment]
Hi @NarkomDe, this is indeed a wrong assumption in the code. Glad you reported it, I'll put out a fix in a little while.

**Comment by frangio** [comment]
@NarkomDe The fix was released in version 1.5.1 of `@openzeppelin/upgrades-core`.

**Comment by NarkomDe** [comment]
Thanks a lot for such a quick update!
